### PR TITLE
fuzz: Use afl++ shared-memory fuzzing

### PR DESCRIPTION
**Pull request by dergoegge**
Using shared-memory is faster than reading from stdin, see https://github.com/AFLplusplus/AFLplusplus/blob/7d2122e0596132f9344a5d0896020ebc79cd33db/instrumentation/README.persistent_mode.md

### CODE REVIEW

**Comment by dergoegge** [comment]
Would be nice to have a afl++ compiler CI task for stuff like this but probably not worth it overall.

**Review by maflcko — APPROVED** [review]
Do you have any number to share on how much difference this makes?

File: src/test/fuzz/fuzz.cpp [hunk 1]
```diff
@@ -197,12 +201,9 @@ int main(int argc, char** argv)
...
     // Enable AFL persistent mode. Requires compilation using afl-clang-fast++.
     // See fuzzing.md for details.
+    const uint8_t* buffer = __AFL_FUZZ_TESTCASE_BUF;
     while (__AFL_LOOP(1000)) {
-        std::vector<uint8_t> buffer;
-        if (!read_stdin(buffer)) {
-            continue;
-        }
-        test_one_input(buffer);
+        test_one_input({buffer, __AFL_FUZZ_TESTCASE_LEN});
```
[the remark this answers is not in the document]
Comment (dergoegge) [reply src/test/fuzz/fuzz.cpp]: Fixed

**Comment by dergoegge** [comment]
> Do you have any number to share on how much difference this makes?

2x according to the [docs](https://github.com/AFLplusplus/AFLplusplus/blob/stable/instrumentation/README.persistent_mode.md#5-shared-memory-fuzzing) but that should subject to the type of disk being used.

I'm seeing a 1.2-1.3x (nvme ssd) running the `miniscript_string` target.

**Review by brunoerg — COMMENTED** [review]
Concept ACK

**Pull request merged**